### PR TITLE
Mbdta 20 - Analytics 4

### PR DESCRIPTION
**Pull request by RyanJField**
Implements Google Analytics 4. **Now** Updated to follow style guidelines.

Imports `www/js/cookies.min.js` to handle JS of cookies.
Imports `www/js/app.js` to bridge JS to Shiny

On accepting the GDPR notice `google_analytics.js` is dynamically loaded substituting the gtag ID into the script.

Declining the GDPR notice stores a cookie stating that analytics has been declined. (This is allowed under essential cookies).

The module determines whether or not to save the cookie in secure mode depending on whether the address in the browser uses https or not, to allow usage with iframes.

When a user changes tab the tab name is sent to google analytics to track usage.

### CODE REVIEW

File: R/analytics_header.R [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+          print("Adding Analytics")
+          # Add the analytics script
+          output$analytics_script <- shiny::renderUI({
+            AddAnalytics(google_analytics_id)
+          })
+        }
+        # Set the Reactive Val to the response
+        is_analytics(as.logical(input$cookie_accept))
+        # Debug Message
+        print("Storing Cookie")
```
Comment (JanionNevill) [inline R/analytics_header.R:257]: Very minor, but could this be "Stored cookie", since it is printed after everything has been done?

**Review by JanionNevill — CHANGES_REQUESTED** [review]
Just minor comments

File: R/analytics_header.R [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+#' @rdname GdprAlert
+#' @importFrom shinyalert shinyalert
+GdprAlert <- function() {
+  shinyalert::shinyalert(
+    title = "GDPR Notice",
+    text = GetGdprMessage(),
+    type = "info",
+    html = TRUE,
+    confirmButtonText = "Accept",
+    showCancelButton = TRUE,
```
[the remark this answers is not in the document]
Comment (RyanJField) [reply R/analytics_header.R:58]: The line below, states that it uses the word "Decline" for the cancel button.

File: R/analytics_header.R (hunk 1 above)
Comment (RyanJField) [reply R/analytics_header.R]: Changed.

File: www/js/js.cookie.min.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
...t)},write:function(e){return encodeURIComponent(e).replace(/%(2[346BF]|3[AC-F]|40|5[BDE]|60|7[BCD])/g,decodeURIComponent)}};return function n(r,o){function i(t,n,i){if("undefined"!=typeof document){"number"==typeof(i=e({},o,i)).expires&&(i.expires=new Date(Date.now()+864e5*i.expires)),i.expires&&(i.expires=i.expires.toUTCString()),t=encodeURIComponent(t).replace(/%(2[346B]|5E|60|7C)/g,decodeURIComponent).replace(/[()]/g,escape),n=r.write(n,t);var c="";for(var u in i)i[u]&&(c+="; "+u,!0!==i[u]&&(c+="="+i[u].split(";")[0]));return document.cookie=t+"="+n+c}}return Object.create({set:i,get:function(e){if("undefined"!=typeof document&&(!arguments.length||e)){for(var n=document.cookie?document.cookie.split("; "):[],o={},i=0;i<n.length;i++){var c=n[i].split("="),u=c.slice(1).join("=");'"'===u[0]&&(u=u.slice(1,-1));try{var f=t.read(c[0]);if(o[f]=r.read(u,f),e===f)break}catch(e){}}return e?o[e]:o}},remove:function(t,n){i(t,"",e({},n,{expires:-1}))},withAttributes:function(t){return n(this.converter,e({},this.attributes,t))},withConverter:function(t){return n(e({},this.converter,t),this.attributes)}},{attributes:{value:Object.freeze(o)},converter:{value:Object.freeze(r)}})}(t,{path:"/"})});
```
[the remark this answers is not in the document]
Comment (RyanJField) [reply www/js/js.cookie.min.js]: JS is minified to reduce the size, this is standard practice. I have updated to the latest release, however this code is incredibly stable. Full source is available: [On the NPM Repo](https://www.npmjs.com/package/js-cookie?activeTab=code) under dist/js.cookie.js or by pasting the code in an [unminify tool](https://unminify.com/).

**Review by tommorris168 — APPROVED** [review]
No more comments from me.

**Review by JanionNevill — APPROVED** [review]
✔

**Pull request merged**